### ISSUE TITLE
Alpha/Speed Lock protection

### ISSUE THREAD

**Issue by Illuminati270** [flybywiresim/aircraft#5094]
### Aircraft Version

Development

### Build info

```json
{
  "built": "2021-06-12T03:56:37+00:00",
  "ref": "master",
  "sha": "476e1ed0d7fa91d0d152a2da762331d25b6067f4",
  "actor": "tracernz",
  "event_name": "manual"
}
```

### Describe the bug

Alpha Lock protection doesn't allow the slats to be retracted when the aircraft is flying high AoAs.

### Expected behavior

Whenever the flap lever is set to 0 with AoA >8.5° and the speed is less than 148kts, the Alpha Lock will avoid slats retraction.

### Steps to reproduce

1. Takeoff with flaps 2 or 3
2. Keep yourself below 148kts, fly a higher than 8° AoA
3. Set flap lever to 0
4. No Alpha Lock activation on ECAM DU 1
5. Slats retracting normally

### References (optional)

![image](https://user-images.githubusercontent.com/85410344/121767053-5fbaa880-cb2c-11eb-9acd-99e72b845b70.png)

DSC-27-30-10

### Discord Username (optional)

Illuminati

**Comment by aguther** [comment]
Some comments to keep in mind when this will be implemented:
- AoA values might not be used 1:1 due to limitations the flight model of MSFS (it needs to be checked how value of 8.5° compares to alpha protection, floor and max; we don't have the real world values of those so that not *that* easy)
- your reference speaks about **slats** only, so I assume this really applies only to slats which cannot be controlled individually
- if it's about not being able to get into CONF 0 but for sure in CONF 1 from higher configurations then this would be possible, but to me it sounds like slats would not retract at all but flaps do

**Comment by Illuminati270** [comment]
I fully understand about the AoA. I did the test with 9° AoA to make sure it wasn't Alpha Locking.

You mentioned that slats can't be controlled individually, but the auto-retraction (>210kts) retracts Flaps from CONF 1+F to CONF 1, actually working perfectly! What if the same code is used with a different rule?

AoA >8.5°
Speed <148kts

- Retract only flaps. (Like when 210kts is reached at CONF 1+F)


And yes, that's how the system works. The slats won't retract (If you have a look on QRH, you'll see that Slats have much more performance impact than the flaps itself).

**Comment by aguther** [comment]
Look, the CONF 1+F to CONF 1 is indeed something we can inhibit and that will comply with what you’ve described by accident because CONF 1+F has slats 18° and flaps 10° while CONF 1 has only the flaps changed to 0°. 

Now let’s talk about CONF 2:
slats = 22° and flaps = 15°

So taking it strict it would mean flaps go to 0° but slats stay at 22°. And that’s something we cannot do right now.

**Comment by Illuminati270** [comment]
Got your point. 

I mean, the system could be partially implemented only for CONF 1 +F, as that would already inhibit someone to select Flaps 0 while flying way below S-speed (on very low weights, of course). Yet, the CONF2/3 -> CONF 0 wouldn't be covered if someone selected by accident (which is mainly why this protection was designed) CONF 0 instead of correct order which would be:

CONF2/CONF3 -> CONF1+F -> CONF 0


Is it a FS2020 restriction right now? Flap/slat jammed is a endless world of possibilities on the Airbus!

**Comment by aguther** [comment]
Yeah that is something we could do as a first step, only inhibit CONF 1 or CONF 1+F to CONF 0 in those conditions.

it would be:
1+F -> 1
1 -> 1

Yes, it’s limitation of MSFS. there is only one lever for slats/flaps…

**Comment by donstim** [comment]
Is it worth trying to pick this up again in terms of only inhibiting the CONF 1 or 1+F to 0 slat retraction?  Or has it already been done?

**Comment by Illuminati270** [comment]
> Is it worth trying to pick this up again in terms of only inhibiting the CONF 1 or 1+F to 0 slat retraction? Or has it already been done?

To this date, it seems not yet implemented. But I believe that's because what Aguther explained on the thread above.

**Comment by FozzieHi** [comment]
Added by #10432
